### PR TITLE
IRV-330: Add React Storybook Example

### PR DESCRIPTION
**Pull request by joemcgill**
Deployed to [styled-components-storybook.irvingjs.com](styled-components-storybook.irvingjs.com)

* Adds React Storybook and configures for use on the Styled Components package.
* Adds (at minimum, except `<App />`) doc scaffolding for all styled components.
* Updates a handful of components to fix bugs discovered during the initial doc scaffolding.

Note: The stories created in this PR are _far_ from perfect. Please keep that in mind. They will be reviewed and updated in subsequent PRs.

### CODE REVIEW

File: packages/styled-components/components/byline/byline.stories.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { PureByline as Byline } from '.';
```
Comment (pattiereaves) [inline packages/styled-components/components/byline/byline.stories.js:3]: oh neat I didn't realize you can just do that with an index file.

File: packages/styled-components/components/byline/byline.stories.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+};
+
+const authors = [
+  <Link href="#">Washington Irving</Link>,
+  <Link href="#">William Irving Sr.</Link>,
+  <Link href="#">Sarah Irving</Link>,
+  <Link href="#">Ebenezer Irving</Link>
+];
+
+export const BasicEample = () => <Byline>{[authors[0]]}</Byline>;
```
Comment (pattiereaves) [inline packages/styled-components/components/byline/byline.stories.js:20]:
```suggestion
export const BasicExample = () => <Byline>{[authors[0]]}</Byline>;
```

File: packages/styled-components/components/socialSharing/socialSharing.stories.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  PureSocialSharing as SocialSharing,
+  SupportedPlatforms,
+} from '.';
+
+export default {
+  component: SocialSharing,
+  title: 'Irving/SocialSharing',
+};
+
+// eslint-next-line max-len
```
Comment (pattiereaves) [inline packages/styled-components/components/socialSharing/socialSharing.stories.js:13]: Does this work? I thought it was 
```suggestion
// eslint-disable-next-line max-len
```
Is it necessary? line 15 looks even longer.

File: packages/styled-components/components/socialSharing/socialSharing.stories.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+const title = "Irving is a ReactJS based, isomorphically rendered, headless CMS frontend application.";
+const description = "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minima vel accusantium odit et minus. Enim quas sint autem consectetur recusandae repellendus veniam natus alias! Ipsam minus, possimus tempora ab illo.";
+const url = "https://irvingjs.org"
+const imageUrl = 'https://placehold.it/1200x630';
+
+export const BasicEample = () => <SocialSharing
```
Comment (pattiereaves) [inline packages/styled-components/components/socialSharing/socialSharing.stories.js:19]:
```suggestion
export const BasicExample = () => <SocialSharing
```

File: packages/styled-components/components/byline/byline.stories.js (hunk 1 above)
Comment (jameswburke) [reply packages/styled-components/components/byline/byline.stories.js:3]: Me neither. @ostowe showed me.

File: packages/styled-components/components/socialSharing/socialSharing.stories.js (hunk 3 above)
Comment (jameswburke) [reply packages/styled-components/components/socialSharing/socialSharing.stories.js]: I guess I assumed it'd be an issue, but the linter is quiet. Removed.

File: package.json [hunk 5]
```diff
@@ -23,14 +23,29 @@
...
+    "deploy-styled-components-storybook": "surge .storybook-styled-components --domain styled-components-storybook.irvingjs.com",
+    "release-storybooks": "npm run build-styled-components-storybook && npm run deploy-styled-components-storybook"
   },
   "dependencies": {
+    "@irvingjs/babel-preset-irving": "^2.14.0",
+    "babel-plugin-module-resolver": "^4.0.0",
     "react": "^16.13.1",
-    "react-dom": "^16.13.1"
+    "react-dom": "^16.13.1",
+    "surge": "^0.21.3"
```
[the remark this answers is not in the document]
Comment (jameswburke) [reply package.json]: Yup. Thanks!

**Pull request merged**